{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae1b4422",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "# -*- coding: utf-8 -*-\n",
    "\"\"\"\n",
    "Created on Wed Apr  3 12:06:30 2024\n",
    "\n",
    "@author: surangijayasinghe\n",
    "\"\"\"\n",
    "\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "import csv\n",
    "import pandas as pd \n",
    "\n",
    "\n",
    "\n",
    "def read_csv(file_path):\n",
    " M5C_positions=[]\n",
    " #row=[]\n",
    " with open(file_path, 'r') as csv_file:\n",
    "  next(csv_file)\n",
    "def read_csv(file_path):\n",
    " M5C_positions=[]\n",
    " #row=[]\n",
    " with open(file_path, 'r') as csv_file:\n",
    "  next(csv_file)\n",
    "  for line in csv_file:\n",
    "     row=line.strip().split(\",\") \n",
    "     position= int(row[0])\n",
    "     #position= data.append(row[0])\n",
    "     #position = int(row[0])\n",
    "     M5C_positions.append(position) \n",
    "\n",
    "     \n",
    " print (\"these are my  positions\", M5C_positions) \n",
    " return  M5C_positions\n",
    "\n",
    "#M5C_positions= read_csv(\"/home/dconklegutierrez/workspace/methylation/m5c/posi>\n",
    "\n",
    "print (\"these are my  positions\", M5C_positions) \n",
    " return  M5C_positions\n",
    "\n",
    "#M5C_positions= read_csv(\"/home/dconklegutierrez/workspace/methylation/m5c/posi>\n",
    "\n",
    "\n",
    "#file_path= (\"/home/dconklegutierrez/workspace/methylation/m5c/positive_control>\n",
    "\n",
    "#csv_data= read_csv(file_path)\n",
    "#for row in csv_data:\n",
    " # print(row)\n",
    "\n",
    "\n",
    "def parse_file(fasta_file, M5C_positions, output_filename):\n",
    "\n",
    "    context_records = []\n",
    "    \n",
    "    def parse_file(fasta_file, M5C_positions, output_filename):\n",
    "\n",
    "    context_records = []\n",
    "  \n",
    "    with open(fasta_file, \"r\") as file:\n",
    "        genome = next(SeqIO.parse(file, \"fasta\"))\n",
    "      \n",
    "        for position in M5C_positions:\n",
    "            #print(\"Postion type:\", type(position)) #ran  into issues  of a str>\n",
    "            context = genome.seq[(int(position - 20)):(int(position + 20))]\n",
    "    \n",
    "            record_id = \"Context_position_\" + str(position)\n",
    "\n",
    "            context_record = SeqRecord(context, id=record_id)\n",
    "\n",
    "            context_records.append(context_record)\n",
    "\n",
    "    with open(output_filename, \"w\") as output_file:\n",
    "        SeqIO.write(context_records, output_file, \"fasta\")\n",
    "        \n",
    "        fasta_file= \"/home/dconklegutierrez/workspace/methylation/m5c/positive_controls>\n",
    "#input_filename = \"/home/dconklegutierrez/workspace/methylation/m5c/positive_co>\n",
    "output_filename = \"output.fasta\"\n",
    "#m5c_positions = [100, 200, 300]  # test\n",
    "#e/methylation/m5c/positive_controls/SRR15720537/SRR15720537_primrose_confident>\n",
    "#M5C_positions= read_csv(\" /home/dconklegutierrez/workspace/methylation/m5c/pos>\n",
    "M5C_positions= read_csv(\"/home/dconklegutierrez/workspace/methylation/m5c/posit>\n",
    "parse_file(fasta_file, M5C_positions, output_filename)\n",
    "\n",
    "print(\"Multifasta file with sequences has been created.\")\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
